layer/filestore.go

```diff
@@ -317,32 +317,38 @@ func (fms *fileMetadataStore) getOrphan() ([]roLayer, error) {
 		}
 
 		for _, fi := range fileInfos {
-			if fi.IsDir() && strings.Contains(fi.Name(), "-removing") {
-				nameSplit := strings.Split(fi.Name(), "-")
-				dgst := digest.NewDigestFromEncoded(algorithm, nameSplit[0])
-				if err := dgst.Validate(); err != nil {
-					logrus.Debugf("Ignoring invalid digest %s:%s", algorithm, nameSplit[0])
-				} else {
-					chainID := ChainID(dgst)
-					chainFile := filepath.Join(fms.root, string(algorithm), fi.Name(), "cache-id")
-					contentBytes, err := ioutil.ReadFile(chainFile)
-					if err != nil {
-						logrus.WithError(err).WithField("digest", dgst).Error("cannot get cache ID")
-					}
-					cacheID := strings.TrimSpace(string(contentBytes))
-					if cacheID == "" {
-						logrus.Errorf("invalid cache id value")
-					}
-
-					l := &roLayer{
-						chainID: chainID,
-						cacheID: cacheID,
-					}
-					orphanLayers = append(orphanLayers, *l)
+			if !fi.IsDir() || !strings.HasSuffix(fi.Name(), "-removing") {
+				continue
+			}
+			encoded := strings.TrimSuffix(fi.Name(), "-removing")
```

Comment:
On this phace `fi.Name()` have value like `<digest>-<random>-removing`. Earlier code did split it to three parts so result was only `<digest>` but as you only trim `-removing` then encoding will have value `<digest>-<random>` so cleanup does not work.

Reply:
@olljanat - modified. PTAL